### PR TITLE
refactor: change default block_size

### PR DESCRIPTION
**Pull request by pphuc25**
Hi,
As mentioned in issue #26069, I have created a new PR with the aim of modifying the block_size for all files. The goal is to set the block size = min(1024, config.max_position_embeddings). This change is intended to ensure synchronization and prevent errors that can occur when the block size exceeds the maximum position embeddings value.
I would like to cc @sanchit-gandhi to review my PR, thank you so much

### CODE REVIEW

**Comment by pphuc25** [comment]
however, I have some confuse because why I can not pass the test case

**Review by ArthurZucker — COMMENTED** [review]
Hey! Running `make fix-copies` should work. 
Also if you plan to do this for a lot of models, make sure to groupe in a single PR.

**Comment by pphuc25** [comment]
thank you @ArthurZucker, but I have some confuse why I just change 1024 to min(1024, config.max_position_embeddings) and docs and then I can not pass the test case, still stuck, can you help me

**Comment by sanchit-gandhi** [comment]
Hey @pphuc25 - don't worry, your PR is perfect! All you need to do is rebase onto `main`:

```
git fetch upstream
git rebase upstream main
```

And then force push:
```
git commit -m "rebase" --allow-empty
git push -f origin flax_min_block_size
```

This should fix your CI issues

**Comment by pphuc25** [comment]
thanks @sanchit-gandhi so much on very helpful information, but this seem to still bug.

File: src/transformers/models/persimmon/modeling_persimmon.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def _expand_mask(mask: torch.Tensor, dtype: torch.dtype, tgt_len: Optional[int]
 
 
 # Copied from transformers.models.llama.modeling_llama.LlamaRotaryEmbedding with Llama->Persimmon
-class PersimmonRotaryEmbedding(torch.nn.Module):
+class PersimmonRotaryEmbedding(nn.Module):
```
Comment (sanchit-gandhi) [inline src/transformers/models/persimmon/modeling_persimmon.py:75]: This change shouldn't be necessary! It's fixed already on `main`. Could you try rebasing one more time please?

Comment (pphuc25) [reply src/transformers/models/persimmon/modeling_persimmon.py:75]: Hmm, seem like nothing change after this debug.

Comment (sanchit-gandhi) [reply src/transformers/models/persimmon/modeling_persimmon.py]: Hey @pphuc25! I checked-out your PR branch, rebased, ran `make fix-copies`, and forced pushed again. This has fixed the copy error we had before.

It does, however, seem to have introduced a few extra code fixes. Were these intended? At a first glance they seem reasonable to me, so we can keep them if so.

**Comment by pphuc25** [comment]
Thank you for really helpful support @sanchit-gandhi, this help me so much and gain me more knowledge, thank you really much.

File: src/transformers/models/persimmon/modeling_persimmon.py (hunk 1 above)
Comment (pphuc25) [reply src/transformers/models/persimmon/modeling_persimmon.py]: I've reinserted them, and it appears that all the processes are functioning smoothly. Once more, I'd like to express my gratitude for your valuable guidance and your kindness, @sanchit-gandhi. Your assistance is greatly appreciated.

**Review by ArthurZucker — COMMENTED** [review]
Actually sorry for the past review, I'll comment the same thing about the `example` folder, which we usually don't maintain. All the improvements that you added are great! But would recommend having your training script shared! This will also bring the spotlight to your work! 🤗

**Comment by pphuc25** [comment]
but @sanchit-gandhi, I think you should review the code to merge PR

**Comment by sanchit-gandhi** [comment]
Shall we merge this one @ArthurZucker since it protects against an edge case, and then add follow-up features to your own codebase / the HF Hub @pphuc25?

**Comment by ArthurZucker** [comment]
Sure!

**Review by sanchit-gandhi — APPROVED** [review]

**Pull request merged**